server/src/test/java/org/elasticsearch/search/query/QueryPhaseCollectorTests.java

```diff
@@ -98,176 +99,143 @@ public static void afterClass() throws Exception {
     }
 
     public void testNullTopDocsCollector() {
-        expectThrows(NullPointerException.class, () -> new QueryPhaseCollector(null, null, 0, null, null));
+        expectThrows(NullPointerException.class, () -> new QueryPhaseCollector(null, null, resolveTerminateAfterChecker(0), null, null));
     }
 
     public void testNegativeTerminateAfter() {
-        expectThrows(
-            IllegalArgumentException.class,
-            () -> new QueryPhaseCollector(new DummyTotalHitCountCollector(), null, randomIntBetween(Integer.MIN_VALUE, -1), null, null)
-        );
-    }
-
-    /**
-     * Collector manager used temporarily to bridge tests and query phase as they have different expectations when it comes to their
-     * return type and how results are retrieved from a collector manager.
-     */
-    private static class CollectorManagerAdapter<C extends Collector, T> implements CollectorManager<Collector, Void> {
-        private final CollectorManager<C, T> wrapped;
-        private T result;
-
-        CollectorManagerAdapter(CollectorManager<C, T> wrapped) {
-            this.wrapped = wrapped;
-        }
-
-        @Override
-        public Collector newCollector() throws IOException {
-            return wrapped.newCollector();
-        }
-
-        @Override
-        public Void reduce(Collection<Collector> collectors) throws IOException {
-            @SuppressWarnings("unchecked")
-            List<C> cs = collectors.stream().map(collector -> (C) collector).toList();
-            result = wrapped.reduce(cs);
-            return null;
-        }
-
-        public T getResult() {
-            return result;
-        }
+        expectThrows(IllegalArgumentException.class, () -> resolveTerminateAfterChecker(randomIntBetween(Integer.MIN_VALUE, -1)));
     }
 
     public void testTopDocsOnly() throws IOException {
         {
             CollectorManager<TopScoreDocCollector, TopDocs> topScoreDocManager = TopScoreDocCollector.createSharedManager(1, null, 1000);
-            CollectorManagerAdapter<TopScoreDocCollector, TopDocs> topScoreDocAdapter = new CollectorManagerAdapter<>(topScoreDocManager);
-            QueryPhaseCollector.CollectorManager manager = QueryPhaseCollector.createManager(topScoreDocAdapter, null, 0, null, null);
-            searcher.search(new MatchAllDocsQuery(), manager);
-            assertFalse(manager.isTerminatedAfter());
-            assertEquals(numDocs, topScoreDocAdapter.getResult().totalHits.value);
+            CollectorManager<QueryPhaseCollector, Result<TopDocs, Void>> manager = createCollectorManager(
```

Comment:
Small suggestion: these might be easier to read with `var`

Reply:
interesting: certainly less typing. Though I need to replace Void with Object somehow. It is less heavy, but is it easier to read given that it's not entirely clear which type I get?

Reply:
I haven't made this change for now. Can do that as a follow-up perhaps.